core/storage/config/gae_models.py

```diff
@@ -180,3 +180,67 @@ def create(
             rules=rule_dicts,
             rule_schema_version=rule_schema_version,
             default_value=default_value)
+
+
+class FeatureFlagModel(base_models.BaseModel):
+    """A class that represents named dynamic feature-flag.
+    This model only stores fields that can be updated in run time.
+
+    The id is the name of the feature-flag.
+    """
+
+    force_enable_for_all_users = datastore_services.BooleanProperty(
+        default=False)
+    rollout_percentage = datastore_services.IntegerProperty(default=0)
+    user_group_ids = datastore_services.JsonProperty(default=[])
+
+    @staticmethod
+    def get_deletion_policy() -> base_models.DELETION_POLICY:
+        """FeatureFlagModel is not related to users."""
+        return base_models.DELETION_POLICY.NOT_APPLICABLE
+
+    @staticmethod
+    def get_model_association_to_user(
+    ) -> base_models.MODEL_ASSOCIATION_TO_USER:
+        """Model does not contain user data."""
+        return base_models.MODEL_ASSOCIATION_TO_USER.NOT_CORRESPONDING_TO_USER
+
+    @classmethod
+    def get_export_policy(cls) -> Dict[str, base_models.EXPORT_POLICY]:
+        """Model doesn't contain any data directly corresponding to a user."""
+        return dict(super(cls, cls).get_export_policy(), **{
+            'force_enable_for_all_users': (
+                base_models.EXPORT_POLICY.NOT_APPLICABLE),
+            'rollout_percentage': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'user_group_ids': base_models.EXPORT_POLICY.NOT_APPLICABLE
+        })
+
+    @classmethod
+    def create(
+        cls,
+        feature_name: str,
+        force_enable_for_all_users: bool,
+        rollout_percentage: int,
+        user_group_ids: List[str]
+    ) -> FeatureFlagModel:
+        """Creates FeatureFlagModel instance.
+
+        Args:
+            feature_name: str. The name of the feature-flag.
+            force_enable_for_all_users: bool. Force enable the feature-flag for
+                all the users.
+            rollout_percentage: int. The defined percentage of logged-in
+                users for which the feature should be enabled.
+            user_group_ids: List[str]. The list of ids of UserGroup objects.
```

Comment:
Where do we validate that these strings are actually valid user group IDs? What happens if a user group ID is subsequently deleted -- will the feature flag model be updated so it doesn't contain invalid data?

Where do we validate that the rollout percentage is between 0 and 100 inclusive? Do we have tests to ensure that invalid values don't get saved and corrupt the datastore?

Reply:
Regarding the user-group IDs, we will be dealing with it in the next part of the project.
The validation for the rollout percentage is present in the domain layer. We only save to the storage after validating the domain object.